src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java

```diff
@@ -336,27 +344,40 @@ public ListBoxModel doFillRepositoryItems(@AncestorInPath SCMSourceOwner context
                     }
                     return result;
                 }
+
+                GHOrganization org = null;
+                try {
+                    org = github.getOrganization(repoOwner);
+                } catch (FileNotFoundException fnf) {
+                    LOGGER.log(Level.FINE, "There is not any GH Organization named " + repoOwner);
+                } catch (IOException e) {
+                    LOGGER.log(Level.WARNING, e.getMessage());
+                }
+                if (org != null && repoOwner.equals(org.getLogin())) {
+                    for (GHRepository repo : org.listRepositories()) {
+                        result.add(repo.getName());
+                    }
+                    return result;
+                }
+
                 GHUser user = null;
                 try {
                     user = github.getUser(repoOwner);
+                } catch (FileNotFoundException fnf) {
+                    LOGGER.log(Level.FINE, "There is not any GH User named " + repoOwner);
                 } catch (IOException e) {
-                    // may be organization... ok to ignore
+                    LOGGER.log(Level.WARNING, e.getMessage());
                 }
                 if (user != null && repoOwner.equals(user.getLogin())) {
-                    for (String name : user.getRepositories().keySet()) {
-                        result.add(name);
+                    for (GHRepository repo : user.listRepositories()) {
+                        result.add(repo.getName());
                     }
                     return result;
                 }
-                GHOrganization org = github.getOrganization(repoOwner);
-                if (org != null && repoOwner.equals(org.getLogin())) {
-                    // TODO this is not what you want; use listRepositories(int) instead
-                    for (String name : org.getRepositories().keySet()) {
-                        result.add(name);
-                    }
-                }
+            } catch (SSLHandshakeException he) {
+                LOGGER.log(Level.SEVERE, he.getMessage());
```

Comment:
@jglick I found this situation when I was working with GH Enterprise and its certicate was not valid. In my opinion, it should be `SEVERE` because you can use the feature. I'd like to include a `Test connection` in global configuration to avoid this, or better, to improve it.